library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -624,91 +632,124 @@ void vtkF3DRenderer::ConfigureHDRI()
     }
 
     this->GetEnvMapPrefiltered()->HalfPrecisionOff();
+    needHDRIComputation = !lutCacheExists || !shCacheExists || !specCacheExists;
```

Comment:
In theory it should not happen, but if `lutCacheExists` is false but `shCacheExists` and `specCacheExists` are true, then you do not need HDRI computation.

```suggestion
    needHDRIComputation = !shCacheExists || !specCacheExists;
```

Reply:
good point, I need to check the logic is fine though

Reply:
are you sure about this ?

```
//------------------------------------------------------------------------------
// Concrete open gl render method.
void vtkOpenGLRenderer::DeviceRender()
{
  vtkTimerLog::MarkStartEvent("OpenGL Dev Render");

  bool computeIBLTextures = !(this->Pass && this->Pass->IsA("vtkOSPRayPass")) &&
    this->UseImageBasedLighting && this->EnvironmentTexture;
  if (computeIBLTextures)
  {
    this->GetEnvMapLookupTable()->Load(this);
    this->GetEnvMapPrefiltered()->Load(this);

    bool useSH = this->UseSphericalHarmonics;

    if (useSH && this->EnvironmentTexture->GetCubeMap())
    {    
      vtkWarningMacro(
        "Cannot compute spherical harmonics of a cubemap, fall back to irradiance texture");
      useSH = false;
    }    

```

to me it looks like a Render is needed, is it not ?